packages/cashscript/src/Transaction.ts

```diff
@@ -169,8 +175,12 @@ export class Transaction {
     hardcodedFee?: number,
     minChange: number = DUST_LIMIT,
     satsPerByte: number = 1.0,
+    _utxos?: Utxo[],
   ): Promise<{ inputs: Utxo[], outputs: OutputForBuilder[] }> {
-    const { utxos } = await this.bitbox.Address.utxo(this.address) as AddressUtxoResult;
+    let utxos = _utxos;
```

Comment:
So when someone passes in their own list of inputs, all of them should be used. I think that with this code change, it only replaces the "pool" of possible UTXOs, but then it still performs the UTXO selection algorithm below. This means that some of the manual inputs might get excluded due to the selection process. 

So instead, if a manual list of utxos is provided, this should completely replace the "selection process". But it should still add a change output where necessary, and it should also check that the provided UTXOs are enough to cover the transaction. Does that make sense?